### PR TITLE
Refactor elements from the core templates

### PR DESCRIPTION
**Pull request by richardhj**
## Description

Add the variable `createEnable` to the MetaModel list template. There is no need to show the `addUrl` button if the table is not creatable.

## Checklist
- [x] Read and understood the [CONTRIBUTING guidelines](CONTRIBUTING.md)
- [ ] Created tests, if possible
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary
- [x] Added myself to the `@authors` in touched PHP files
- [ ] Checked the changes with phpcq and introduced no new issues

### CODE REVIEW

**Comment by zonky2** [comment]
you can switch the year to `2017` ;-)

**Comment by discordier** [comment]
Nice spot, however creating a new instance of DCG for every rendering iteration is a pretty heavy lift.

We should improve on that one.

**Comment by richardhj** [comment]
But the `RenderItemListEvent` is called per list and not per item, isn't it?
But true, I was not sure whether this is the most efficient way to do this.

**Comment by discordier** [comment]
I guess we now need the custom template as discussed in MetaModels/core#1078 here also?

**Comment by richardhj** [comment]
The frontend-editing module uses the `Contao\BaseTemplate` instead of the `MetaModels\Render\Template` to render the templates. (Maybe) This is why ```<?php $this->extend('ce_metamodel_list'); ?>``` does not work in the `ce_metamodel_list` template in the frontend-editing module.
Is there a way to change the rendering class?

**Comment by discordier** [comment]
I can not follow you on this one, in the fe-module, the `ce_metamodel_list` is not used at all, so why should it extend there?

**Comment by richardhj** [comment]
Have a look at 752e2f6 then please

File: contao/config/autoload.php [hunk 1]
```diff
@@ -25,4 +25,8 @@
 (
     'ce_metamodel_frontend_edit'  => 'system/modules/metamodels-contao-frontend-editing/templates',
     'mod_metamodel_frontend_edit' => 'system/modules/metamodels-contao-frontend-editing/templates',
+    'ce_metamodel_list'           => 'system/modules/metamodels-contao-frontend-editing/templates',
+    'mod_metamodel_list'          => 'system/modules/metamodels-contao-frontend-editing/templates',
+    'metamoel_prerendered'        => 'system/modules/metamodels-contao-frontend-editing/templates',
```
Comment (discordier) [inline contao/config/autoload.php]: As pointed out by @zonky2, there is a `d` missing here. :smile_cat:

**Comment by discordier** [comment]
Now I get it... of course the templates must all be named `ce_metamodel_listedit` or something like that. Otherwise the extending of the templates would end up in an endless loop.
A template can not extend itself but only another (regarding name) template.

**Comment by richardhj** [comment]
@discordier Please support providing variable `isCreatable` as mentioned in https://github.com/MetaModels/contao-frontend-editing/pull/7#issuecomment-270726630

**Comment by richardhj** [comment]
Templates are fine. Next step could be adding an event listener to set the `ce_metamodel_list_edit` per default when enabling `metamodel_edit` in the dca.

**Comment by richardhj** [comment]
Remove the `isCreatable` variable to make DCG instance creation unnecessary // https://github.com/MetaModels/contao-frontend-editing/pull/7#issuecomment-270726630
RTM

**Comment by discordier** [comment]
Fine, now squash them and we are ready to go.

**Comment by richardhj** [comment]
👌

**Comment by zonky2** [comment]
welcome OWD 👌

**Comment by discordier** [comment]
@richardhj I have just merged MetaModels/core#1078, therefore you can proceed here now at will.

**Pull request merged**